resolver/auth_mutations.resolvers_test.go

```diff
@@ -614,155 +468,253 @@ func TestChangePassword(
+				AddRow(1, testutls.MockEmail, testutls.MockToken, 1)
+			mock.ExpectQuery(regexp.QuoteMeta(`SELECT "users".* FROM "users" WHERE (token=$1) LIMIT 1;`)).
+				WithArgs().
+				WillReturnRows(rows)
+		},
+	}
 }
 
-func prepareMockJWTService(name string) *gomonkey.Patches {
-	return gomonkey.ApplyFunc(service.JWT, func(cfg *config.Configuration) (jwt.Service, error) {
-		if name == ErrorFromJwt {
-			return jwt.Service{}, fmt.Errorf(ErrorMsgFromJwt)
-		}
-		return jwt.Service{}, nil
-	})
+func refreshTokenSuccessCase() refereshTokenType {
+	return refereshTokenType{
+		name: SuccessCase,
+		req:  ReqToken,
+		wantResp: &fm.RefreshTokenResponse{
+			Token: testutls.MockToken,
+		},
+		wantErr: false,
+		init: refereshTokenPatches{
+			configPatch: func() *gomonkey.Patches {
+				return gomonkey.ApplyFunc(config.Load, func() (*config.Configuration, error) {
+					return &config.Configuration{}, nil
+				})
+			},
+			jwtPatch: func() *gomonkey.Patches {
+				tg := jwt.Service{}
+				return gomonkey.ApplyFunc(service.JWT, func(cfg *config.Configuration) (jwt.Service, error) {
+					return tg, nil
+				})
+			},
+			tokenPatch: func() *gomonkey.Patches {
+				tg := jwt.Service{}
+				return gomonkey.ApplyMethod(reflect.TypeOf(tg), "GenerateToken",
+					func(jwt.Service, *models.User) (string, error) {
+						return "token", nil
+					})
+			},
+		},
+		initMocks: func(mock sqlmock.Sqlmock) {
+			rows := sqlmock.NewRows([]string{"id", "email", "token", "role_id"}).
+				AddRow(1, testutls.MockEmail, testutls.MockToken, 1)
+			mock.ExpectQuery(regexp.QuoteMeta(`SELECT "users".* FROM "users" WHERE (token=$1) LIMIT 1;`)).
+				WithArgs().
+				WillReturnRows(rows)
+		},
+	}
 }
-
-func prepareMockTokenGeneration(name string) *gomonkey.Patches {
-	return gomonkey.ApplyMethod(reflect.TypeOf(jwt.Service{}), "GenerateToken",
-		func(jwt.Service, *models.User) (string, error) {
-			if name == ErrorFromGenerateToken {
-				return "", resultwrapper.ErrUnauthorized
-			}
-			return "token", nil
-		})
+func loadRefereshTokenCases() []refereshTokenType {
+	return []refereshTokenType{
+		refreshTokenInvalidCase(),
+		refreshTokenErrorFromConfigCase(),
+		refereshTokenErrorFromJWTCase(),
+		refereshTokenErrorFromGenerateTokenCase(),
+		refreshTokenSuccessCase(),
+	}
 }
 
-func prepareContextRefereshToken() context.Context {
-	c := context.Background()
-	return context.WithValue(c, testutls.UserKey, testutls.MockUser())
+func TestRefreshToken(t *testing.T) {
+	cases := loadRefereshTokenCases()
+	// Create a new instance of the resolver
+	resolver1 := resolver.Resolver{}
+	for _, tt := range cases {
+		t.Run(
+			tt.name,
+			func(t *testing.T) {
+				// Create a mock SQL database connection
+				mock, cleanup, _ := testutls.SetupMockDB(t)
+				defer cleanup()
+				// Handle the case where authentication token is invalid
+				tt.initMocks(mock)
+				// Handle the case where there is an error loading the config
+				configpatch := tt.init.configPatch()
```

Comment:
instead we can just return all 3 patches in a single init call. So we don't need to know the implementation details of init. 

Same like the other places